src/components/ui_kit/activity_indicator.rs

```diff
@@ -1,23 +1,40 @@
 use dioxus::prelude::*;
+use warp::{crypto::{DID}, multipass::identity::IdentityStatus};
+
+use crate::Account;
 
 #[derive(PartialEq, Props)]
 pub struct Props {
     inline: bool,
+    remote_did: DID,
+    account: Account,
 }
 
 #[allow(non_snake_case)]
 pub fn ActivityIndicator(cx: Scope<Props>) -> Element {
+    let identity = &cx.props.account;
+    
+    let status = match identity.read().identity_status(&cx.props.remote_did) {
+        Ok(s) => s,
+        Err(_) => IdentityStatus::Offline,
+    };
+
     let main_class = match cx.props.inline {
         true => "inline",
         false => "icon-icon",
     };
 
+    let bubble_class = match status {
```

Comment:
since `IdentityStatus` implements `Display` you should be able to do `let buble_class = status.to_string();` or even just have it apart of the div by doing `div { class: "bubble {status}" }` (assuming the ui lib allows it, which i believe it does for anything implementing `Display`)